File: c++/5 kyu Consecutive k-Primes.cpp
#include <vector>
using namespace std; 
class PrimeConsec
{
private:
    static int countPrimeDiv(long long n){
    int c=0;
		for(long long i=2;i<=n;i++)
		while(n%i==0)
    {
		  n/=i;
		  c++;
	   }	
		return c;
	}
public:
  static int consecKprimes(int k, std::vector<long long> &arr){
  int c=0;
	for(int i=0;i<arr.size()-1;i++)
	if(countPrimeDiv(arr[i])==countPrimeDiv(arr[i+1])&&countPrimeDiv(arr[i+1])==k)  c++;
	return c;
	}
};

File: Kotlin/5 kyu Consecutive k-Primes.kt
package conseckprimes

fun consecKprimes(k:Int, arr:LongArray):Int {
   var c=0
    if(arr.size==1)return 0
	for(i in 0..arr.size-2)
        if(Check(arr[i])==k && k==Check(arr[i+1])) c++
    return c;
}
fun Check(n1:Long):Int{
	var n=n1;
	var c=0
	for(j in 2..n)
    while(n % j == 0.toLong()){
		c++;
		n/=j;
      	}
	return c
}
